price-feeder/oracle/provider/binance.go

```diff
@@ -23,96 +26,187 @@ type (
 	// BinanceProvider defines an Oracle provider implemented by the Binance public
 	// API.
 	//
-	// REF: https://github.com/binance/binance-spot-api-docs/blob/master/rest-api.md
+	// REF: https://binance-docs.github.io/apidocs/spot/en/#individual-symbol-mini-ticker-stream
 	BinanceProvider struct {
-		baseURL string
-		client  *http.Client
+		wsURL           url.URL
+		wsClient        *websocket.Conn
+		logger          zerolog.Logger
+		mu              sync.Mutex
+		tickers         map[string]BinanceTicker // Symbol => BinanceTicker
+		subscribedPairs []types.CurrencyPair
 	}
 
-	// BinanceTickerResponse defines the response structure of a Binance ticker
-	// request.
-	BinanceTickerResponse struct {
-		Symbol    string `json:"symbol"`
-		LastPrice string `json:"lastPrice"`
-		Volume    string `json:"volume"`
+	// BinanceTicker ticker price response
+	// https://pkg.go.dev/encoding/json#Unmarshal
+	// Unmarshal matches incoming object keys to the keys
+	// used by Marshal (either the struct field name or its tag),
+	// preferring an exact match but also accepting a case-insensitive match
+	// C is not used, but it avoids to implement specific UnmarshalJSON
+	BinanceTicker struct {
+		Symbol    string `json:"s"` // Symbol ex.: BTCUSDT
+		LastPrice string `json:"c"` // Last price ex.: 0.0025
+		Volume    string `json:"v"` // Total traded base asset volume ex.: 1000
+		C         uint64 `json:"C"` // Statistics close time
+	}
 
-		// Code and Msg are populated on failed requests
-		Code int    `json:"code"`
-		Msg  string `json:"msg"`
+	// BinanceSubscribeMsg Msg to subscribe all the tickers channels
+	BinanceSubscribeMsg struct {
```

Comment:
nit: just for consistency with OkxSubscriptionMsg
```suggestion
	BinanceSubscriptionMsg struct {
```